assets/js/blocks/product-query/variations/product-query.tsx

```diff
@@ -65,16 +64,24 @@ const registerProductsBlock = ( attributes: QueryBlockAttributes ) => {
 };
 
 if ( isWpVersion( '6.1', '>=' ) ) {
-	let currentTemplateId: string | undefined;
-	subscribe( () => {
-		const previousTemplateId = currentTemplateId;
-		const store = select( 'core/edit-site' );
-		currentTemplateId = store?.getEditedPostId();
-		if ( previousTemplateId === currentTemplateId ) {
-			return;
-		}
+	const store = select( 'core/edit-site' );
+	const isTemplate = store?.getEditedPostType() === 'wp_template';
+
+	if ( isTemplate ) {
+		let currentTemplateId: string | undefined;
+
+		subscribe( () => {
+			const previousTemplateId = currentTemplateId;
+
+			currentTemplateId = store?.getEditedPostId();
+
+			if (
+				previousTemplateId === currentTemplateId ||
+				typeof currentTemplateId === 'undefined'
+			) {
+				return;
+			}
```

Comment:
The current logic has a heavy assumption on `isTemplate` either being fully initialized when `registerProductsBlock` is called or never changing in the app's lifecycle. That's not a good assumption to make here. Why not have this check within the subscribe callback?

Reply:
Good point. I refactored this part of the code in 2057f0f920e31dd6e4f1ffebf7f322c798b28bae, I think it should be more resilient now and (slightly) more performant.